File: backend/app/schemas/ai.py
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field


class AiChatRequest(BaseModel):
    message: str = Field(min_length=1, max_length=1200)
    session_id: str | None = None
    application_id: UUID | None = None
    current_step: int | None = Field(default=None, ge=1, le=5)


class AiAssistantSource(BaseModel):
    topic_id: str
    title: str
    source_label: str
    source_path: str
    excerpt: str


class AiChatResponse(BaseModel):
    answer: str
    session_id: str
    mode: Literal["rules", "langchain"]
    sources: list[AiAssistantSource]
    suggested_prompts: list[str]

File: frontend/src/types/ai.ts
export type AiAssistantSource = {
  topic_id: string;
  title: string;
  source_label: string;
  source_path: string;
  excerpt: string;
};

export type AiChatRequest = {
  message: string;
  session_id?: string;
  application_id?: string;
  current_step?: number;
};

export type AiChatResponse = {
  answer: string;
  session_id: string;
  mode: "rules" | "langchain";
  sources: AiAssistantSource[];
  suggested_prompts: string[];
};
